server/api/annotation.py

```diff
@@ -68,66 +73,102 @@ def find(self, params):
-            '_id': annotation['_id'],
-            '_modelType': 'annotation',
-            'studyId': annotation['meta']['studyId'],
-            'image': Image().filterSummary(
-                Image().load(annotation['meta']['imageId'], force=True, exc=True),
-                currentUser),
-            'user': User().filterSummary(
-                user=User().load(annotation['meta']['userId'], force=True, exc=True),
-                accessorUser=currentUser),
-            'state': Annotation().getState(annotation)
-        }
-        if Annotation().getState(annotation) == Study().State.COMPLETE:
-            output.update({
-                'annotations': annotation['meta']['annotations'],
-                'status': annotation['meta']['status'],
-                'startTime': annotation['meta']['startTime'],
-                'stopTime': annotation['meta']['startTime'],
-            })
-
-        return output
+        user = self.getCurrentUser()
+        return Annotation().filter(annotation, user)
+
+    @describeRoute(
+        Description('Return an annotation feature as a raw superpixel array.')
+        .param('annotationId', 'The ID of the annotation.', paramType='path')
+        .param('featureId', 'The feature ID within the annotation.', paramType='path')
+        .errorResponse()
+    )
+    @access.cookie
+    @access.public
+    @loadmodel(map={'annotationId': 'annotation'}, model='annotation', plugin='isic_archive',
+               level=AccessType.READ)
+    def getAnnotationFeature(self, annotation, featureId, params):
+        study = Study().load(annotation['meta']['studyId'], force=True, exc=True)
+        featureset = Study().getFeatureset(study)
+
+        if not any(featureId == feature['id'] for feature in featureset['localFeatures']):
+            raise ValidationException('Invalid featureId.', 'featureId')
+        if Annotation().getState(annotation) != Study().State.COMPLETE:
+            raise RestException('Only complete annotations have superpixel data.')
+
+        featureValues = annotation['meta']['annotations'].get(featureId, [])
+        return featureValues
+
+    @describeRoute(
+        Description('Return an annotation feature as a mask.')
+        .param('annotationId', 'The ID of the annotation.', paramType='path')
+        .param('featureId', 'The feature ID within the annotation.', paramType='path')
+        .produces('image/png')
+        .errorResponse()
+    )
+    @access.cookie
+    @access.public
+    @loadmodel(map={'annotationId': 'annotation'}, model='annotation', plugin='isic_archive',
+               level=AccessType.READ)
+    def getAnnotationFeatureMask(self, annotation, featureId, params):
+        study = Study().load(annotation['meta']['studyId'], force=True, exc=True)
+        featureset = Study().getFeatureset(study)
+
+        if not any(featureId == feature['id'] for feature in featureset['localFeatures']):
+            raise ValidationException('Invalid featureId.', 'featureId')
```

Comment:
It may be of interest that, when `produces` is defined above, swagger-ui never indicates this error response, even though the request completes.

Reply:
Reported upstream https://github.com/girder/girder/issues/2553